ravitandon90/ThreeHundredDaysOfCode-FE
# [Refactor] Include latest changes from upstream

This CL includes changes to -
1. Introduce session-context for managing auth state. (handles logout/login)
2. use reducers and dispatchers to update the session-context's state.
3. SearchBar - make filteredSuggestions and activeSuggestion local instead of in app context.
(Context is used for storing data which changes infrequently).

TODO -
1. make code less verbose
2. use a separate library for API calls ("/src/lib/api")

TEST -
1. login/logout -  (non)Incognito, SSO, basicAuth modes
2. Check UI design for any inconsistencies.
